esx_service/utils/auth_data.py

```diff
@@ -0,0 +1,306 @@
+        for p in default_privileges:
+            p['tenant_id'] = id
+        self.conn.executemany(
+            """
+            insert into privileges values
+            (:tenant_id, :datastore, :global_visibility, :create_volume,
+             :delete_volume, :mount_volume, :max_volume_size, :usage_quota)
+            """,
+            default_privileges
+        )
+
+        for p in privileges:
+            p['tenant_id'] = id
+        self.conn.executemany(
+            """
+            insert into privileges values
+            (:tenant_id, :datastore, :global_visibility, :create_volume,
+             :delete_volume, :mount_volume, :max_volume_size, :usage_quota)
+            """,
+            privileges
+        )
+        self.conn.commit()
+        return tenant
+    
+    def list_tenants(self):
+        """
+            Return a list of all tenants
+        """
+        tenant_list = []
+        cur = self.conn.execute(
+           "select * from tenants"
+        )
+        result = cur.fetchall()
+       
+        for r in result:
+            # loop through each tenant
+            id = r[0]
+            name = r[1]
+            description = r[2]
+            default_datastore = r[3]
+            
+            # search vms for this tenant
+            vms = []
+            cur = self.conn.execute(
+                "select * from vms where tenant_id=?",
+                (id,)
+            )
+            vms = cur.fetchall()
+            
+            # search privileges for this tenant
+            privileges = []
+            cur = self.conn.execute(
+                "select * from privileges where tenant_id=? and datastore!=?",
+                (id,default_datastore)    
+            )
+            privileges = cur.fetchall()
+
+            default_privileges = []
+            cur = self.conn.execute(
+                "select * from privileges where tenant_id=? and datastore=?",
+                (id,default_datastore)    
+            )
+            default_privileges = cur.fetchall()
+            #print "default_privileges"
+            #print default_privileges
+            tenant = DockerVolumeTenant(name, description, default_datastore,
+                                        default_privileges, vms, privileges, id)
+            tenant_list.append(tenant)
+
+        return tenant_list   
+
+    def remove_tenant(self, tenant_id, remove_volumes):
+        """
+            Remove a tenant with given id
+            A row with given tenant_id will be removed from table tenants, vms, 
+            and privileges
+            All the volumes created by this tenant will be removed if remove_volumes
+            is set to True 
+        """
+      
+        self.conn.execute(
+                "delete from tenants where id=?", 
+                [tenant_id]
+        )
+
+        self.conn.execute(
+                "delete from vms where tenant_id=?", 
+                [tenant_id]
+        )
+
+        self.conn.execute(
```

Comment:
would it be better to remove this first to make sure there are no dangling foreign keys ?